Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.Add.cs

```diff
@@ -0,0 +1,66 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Moq;
+using Taarafo.Core.Models.Profiles;
+using Taarafo.Core.Models.Profiles.Exceptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.Profiles
+{
+    public partial class ProfileServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowCriticalDependencyExceptionOnAddIfSqlErrorOccursAndLogItAsync()
+        {
+            // given
+            DateTimeOffset dateTime = GetRandomDateTime();
+            Profile randomProfile = CreateRandomProfile(dateTime);
+            SqlException sqlException = GetSqlException();
+
+            var failedProfileStorageException =
+                new FailedProfileStorageException(sqlException);
+
+            var expectedProfileDependencyException =
+                new ProfileDependencyException(failedProfileStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Returns(dateTime);
```

Comment:
Why not just throw the `sqlException` from here since we are not really trying to validate the flow of events but rather how we act when these events occur.

Reply:
done